internal/stacks/stackconfig/config.go

```diff
@@ -297,6 +323,17 @@ func loadConfigDir(sourceAddr sourceaddrs.FinalSource, sources *sourcebundle.Bun
 				continue
 			}
 
+			if source == nil {
+				source = effectiveSourceAddr
```

Comment:
There does not seem to be any use of this assignment.

Could this entire conditional not be

```
if source != nil && source != effectiveSourceAddr {
	diags = diags.Append(&hcl.Diagnostic{
		Severity: hcl.DiagError,
		Summary:  "Invalid source address",
		Detail:   fmt.Sprintf("Cannot use %q as a source address here: the target stack is already initialised with another source %q.", effectiveSourceAddr, source),
		Subject:  rmvd.SourceAddrRange.ToHCL().Ptr(),
	})
}
```

Reply:
If a removed block targets a component that doesn't exist in the configuration anymore, then `source` will be nil. In that case, we assign `source` to be the first `effectiveSourceAddr` that we discover from the removed blocks, and future removed blocks are validated against that source. If the component does still exist in the configuration then source is not nil and all the removed blocks are validated against that value.

On lines 304 - 309, you can see that `source` is only set conditionally.

Reply:
Oh!. I missed the for-loop.